### PR TITLE
PLAT-142922: Input: Add tel, passwordtel type

### PR DESCRIPTION
**Pull request by changgilee**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [ ] Documentation was added or is not needed
* [x] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Need tel type VKB for input number and special characters

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Add tel, passwordtel type in Input component

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
N/A

### Links
[//]: # (Related issues, references)
https://jira2.lgsvl.com/browse/PLAT-142922

### Comments
N/A

### CODE REVIEW

File: Input/Input.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ import {DEFAULT_LENGTH, calcAriaLabel, convertToPasswordFormat, extractInputFiel
 import componentCss from './Input.module.less';
 
 const prepareInputEventPayload = ev => ({value: ev.target.value});
-const isPasswordType = type => type === 'password' || type === 'passwordnumber';
+const isPasswordType = type => type === 'password' || type === 'passwordnumber' || type === 'passwordtel';
```
Comment (MikyungKim) [inline Input/Input.js:26]:
```suggestion
const isPasswordType = type => type.includes('password');
```

File: Input/InputField.js [hunk 2]
```diff
@@ -291,13 +292,13 @@ const InputFieldBase = kind({
 				<input
 					{...inputProps}
 					{...voiceProps}
-					className={css.input}
+					className={type === 'passwordtel' ? classnames(css.input, css.passwordtel) : css.input}
```
Comment (MikyungKim) [inline Input/InputField.js:295]: Why don't we use `computed` block and define `inputClassName`?

Comment (MikyungKim) [reply Input/InputField.js:295]: inputClassName: ({styler, type}) => styler.append('input', {passwordtel: (type === 'passwordtel')}),

File: Input/Input.js (hunk 1 above)
Comment (changgilee) [reply Input/Input.js]: Fix it. Thanks~

File: Input/InputField.js (hunk 2 above)
Comment (changgilee) [reply Input/InputField.js]: Fix it

**Review by 0x64 — APPROVED** [review]
:lgtm:

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,13 @@
 
 The following is a curated list of changes in the Enact sandstone module, newest changes on the top.
 
+## [unreleased]
+
+### Added
+
+- `sandstone/Input` type `tel` and `passwordtel`
+
```
Comment (MikyungKim) [inline CHANGELOG.md:10]: Please remove the one empty line.

Comment (changgilee) [reply CHANGELOG.md:10]: Fix it

File: Input/Input.js [hunk 4]
```diff
@@ -441,7 +441,7 @@ const InputBase = kind({
 		 * @default 'text'
```
Comment (MikyungKim) [inline Input/Input.js:441]: Please add `tel` and `passwordtel` at line 440.

File: Input/InputField.js [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
 import {I18nContextDecorator} from '@enact/i18n/I18nDecorator';
 import {isRtlText} from '@enact/i18n/util';
 import Changeable from '@enact/ui/Changeable';
 import Pure from '@enact/ui/internal/Pure';
 import compose from 'ramda/src/compose';
 import PropTypes from 'prop-types';
 import React from 'react';
+import {useAnnounce} from '@enact/ui/AnnounceDecorator';
+import {readAlert} from '@enact/webos/speech';
+import platform from '@enact/core/platform';
```
Comment (MikyungKim) [inline Input/InputField.js:12]: Please re-order imports by packagename.

File: Input/InputField.js [hunk 6]
```diff
@@ -33,6 +36,14 @@ const InputFieldBase = kind({
 	name: 'InputField',
 
 	propTypes: /** @lends sandstone/Input.InputFieldBase.prototype */ {
+		/**
+		 * Passed by AnnounceDecorator for accessibility.
+		 *
+		 * @type {Object}
```
Comment (MikyungKim) [inline Input/InputField.js:42]:
```suggestion
		 * @type {Function}
```

File: Input/InputField.js [hunk 7]
```diff
@@ -291,14 +313,15 @@ const InputFieldBase = kind({
...
+					aria-hidden={type === 'passwordtel'}
+					className={inputClassName}
 					dir={dir}
 					disabled={disabled}
 					onChange={onChange}
 					placeholder={placeholder}
 					tabIndex={-1}
-					type={type}
 					value={value}
+					type={type === 'passwordtel' ? 'tel' : type}
```
Comment (MikyungKim) [inline Input/InputField.js:324]: Let's put this line after `tabIndex`.

File: Input/InputField.js [hunk 8]
```diff
@@ -257,6 +277,7 @@ const InputFieldBase = kind({
 		},
 		className: ({invalid, size, styler}) => styler.append({invalid}, size),
 		dir: ({value, placeholder}) => isRtlText(value || placeholder) ? 'rtl' : 'ltr',
+		inputClassName: ({styler, type}) => styler.append('input', {'passwordtel': (type === 'passwordtel')}),
```
Comment (MikyungKim) [inline Input/InputField.js:280]:
```suggestion
		inputClassName: ({styler, type}) => styler.append('input', {passwordtel: (type === 'passwordtel')}),
```

Comment (changgilee) [reply Input/InputField.js]: Fix all things~ Thanks

File: Input/InputField.js [hunk 9]
```diff
@@ -1,9 +1,12 @@
 import kind from '@enact/core/kind';
-import {handle, adaptEvent, forwardCustom, forwardWithPrevent} from '@enact/core/handle';
+import platform from '@enact/core/platform';
+import {handle, adaptEvent, forwardCustom, forwardWithPrevent, returnsTrue} from '@enact/core/handle';
```
Comment (MikyungKim) [inline Input/InputField.js:3]: Let's put this at line 1.

**Pull request merged**